chapter4/astroml_chapter4_Gaussian_mixtures.ipynb

```diff
@@ -0,0 +1,363 @@
+    "\n",
+    "We'll focus our discussion on Gaussian mixtures, and assume that the measurement errors follow Gaussian distributions with standard deviations $e_i$. In the case of **homoscedastic** error where all $e_i = e$, we can rely on the fact that the convolution of two Gaussians is a Gaussian and obtain \"intrinsic \"widths\n",
+    "\n",
+    "$$\\sigma^*_j = (\\sigma_j^2 - e^2)^{1 / 2}.$$\n",
+    "\n",
+    "This correction procedure fails in the **heteroscedastic** case. Additionally, due to uncertainties in the best-fit values, it's possible that the best-fit value of $\\sigma_j$ may turn out to be smaller than $e$.\n",
+    "\n",
+    "A remedy is to account for measurement errors already in the model description: we can replace $\\sigma_j$ in eq. $(1)$ by $(\\sigma_j^2 + e_i^2)^{1/2}$, where the $\\sigma_j$ now correspond to the intrinsic widths of each class. However, these new class pdfs **do not admit simple explicit prescriptions for the maximization step given by Eqs. 4 – 5 since the Gaussian has a different width for each $x_i$**\n",
+    "\n",
+    "Following the same derivation steps, the new prescriptions for the M-step are now\n",
+    "\n",
+    "$$\\mu_j = \\frac{\\sum^N_{i=1} \\frac{w_{ij}}{\\sigma_j^2 +e_i^2} x_i}{\\sum^N_{i=1} \\frac{w_{ij}}{\\sigma_j^2 +e^2_i}} $$\n",
+    "\n",
+    "and\n",
+    "\n",
+    "$$\\qquad \\qquad \\sum^N_{i=1} \\frac{w_{ij}}{\\sigma_j^2 - e_i^2} = \\sum^N_{i=1} \\frac{w_{ij}}{(\\sigma^2_j + e^2_i)^2} (x_i - \\mu_j)^2. \\qquad \\qquad (7)$$\n",
+    "\n",
+    "\n",
+    "Compared to Eqs. 4 – 5, $\\sigma_j$ is now \"coupled\" to $e_i$ and cannot be moved outside of the sum, which prevents a few cancelations that led to simple forms when $e_i = 0$. Update rules for $\\mu_j$ and $\\alpha_j$ are still explicit and would require only a minor modification of specific implementation. The main difficulty, which prevents the use of standard EM routines for performing the M-step, is eq. $7$, because the update rule for $\\sigma_j$ is not explicit anymore. Nevertheless, it still provides a rule for updating $\\sigma_j$, which can be found by numerically solving eq. $7$.\n",
+    "\n",
+    "##### Non-Gaussian mixture models\n",
+    "\n",
+    "The EM algorithm is not confined to Gaussian mixtures. As eq. $3$ shows, the basic premise of the method can be derived for any mixture model. In addition to various useful properties discussed in §3.3.2, a major benefit of Gaussian pdfs is the very simple set of explicit equations (Eqs. 4 – 6) for updating model parameters. When other pdfs are used, a variety of techniques are proposed in the literature for implementation of the maximization M-step. For cases where Gaussian mixtures are insufficient descriptors of data, we recommend consulting abundant and easily accessible literature on the various forms of the EM algorithm."
```

Comment:
We'll need to fix the references to point to other notebook sections rather than refer back to the textbook that the target audience would not have access to.

I spotted e.g. this one first, but there are other examples, too.
I open an issue for this as this is not a big enough problem to hold up the PR for, and also I suspect this would need to be systematically go through of all the notebooks.